gammapy/modeling/models/core.py

```diff
@@ -163,6 +163,9 @@ def parameters(self):
             [getattr(self, name) for name in self.default_parameters.names]
         )
 
+    def stat_sum(self):
```

Comment:
I think you are right.  To me it seems that a single model is a subset of `Models` (e.g. `len(Models) == 1`) right?  Then this would surely be better placed there.  Please forgive my ignorance about the Gammapy internals and where put this best.

Reply:
I think we should figure out a clean API here. I think the `Models.stat_sum()` is confusing a bit as well. Long-term we would like to have a more separated API between `MapDataset`, `Models` and `Fit`. I am very sure that we will evolve into a direction where models are passed to e.g `MapDataset.stat_sum(models=models)`. So I would propose to evolve into a direction of e.g. implementing a `PriorFitStatistic` class, which takes the `Models` class and evaluates the priors. Like `PriorFitStatistic(beta=1.).stat_sum(models=models)`. By now it seems to me this `PriorFitStatistics` will only need to supported for `Datasets`.